lib/services/salesforce.rb

```diff
@@ -11,20 +11,28 @@ class AhaServices::Salesforce < AhaService
   
   install_button
 
+  def receive_updated
+    validate_host
+  end
+
   def receive_installed
-    # Validate authentication.
-    client.user_info
-    
-    # Update settings.
-    client.post('/services/apexrest/ahaapp/aha_rest_api/settings', 
-      idea_portal_url: data.idea_portal_url, 
-      jwt_secret_key: data.jwt_secret_key)
-  rescue Exception => e
-    logger.debug("Salesforce authentication problem #{e.class}: #{e.message} #{e.backtrace.join("\n")}")
-    if e.message.include? 'The REST API is not enabled for this Organization.'
-      raise ConfigurationError, "The REST API is not enabled for this Organization."
-    else
-      raise ConfigurationError, "Authentication failed. Please use the 'Authenticate' button to connect to Salesforce."
+    validate_host or raise ConfigurationError, "Please ensure the custom host is a host ending in salesforce.com."
```

Comment:
I think we should make this error message more clear, a user who entered `https://myurl.salesforce.com` probably does not understand from this message that a "host" doesn't include the `https://` part. I see we're putting it in the error logs but it seems to me that it should show in the error msg as well. I think it'd be fine if we returned the error strings being logged in `validate_host` and then made the error message `"Invalid host: #{error_msg}"` (or we could just raise the `ConfigurationError` directly within `validate_host`)